### ISSUE TITLE
Generating italic TTFs messes up spacing.

### ISSUE THREAD

**Issue by ryanbugden** [adobe-type-tools/afdko#1516]
I think something is happening with FDK’s interpretation of extrema on slanted shapes, which is drastically compromising spacing.

I've made two nearly identical test UFOs with the glyph `c`. The other glyphs are just reference blocks. The only difference between these two UFOs is that one has conventional "upright" extreme points, which solves the spacing issue.

Please see the UFOs and their resulting TTFs below:
[test.zip](https://github.com/adobe-type-tools/afdko/files/8904516/test.zip)

**Comment by jenskutilek** [comment]
It seems makeotf puts into the `hmtx` table the minimum x coordinate that the curve touches, not any actual point on or off the curve:

```xml
<mtx name="c" width="626" lsb="23"/>
```

The leftmost off-curve point in c is at x = 2, I think that’s what should go into the `lsb` value (and changing it manually indeed fixes the spacing).

**Comment by frankrolf** [comment]
Ha, unfortunately `makeotf` can really only make OTFs :-)
The -o option is just for moving files around.

We (in Source Serif, for example) make TTF files using `otf2ttf`, a fontTools/misc script.

**Comment by justvanrossum** [comment]
Ha, it crashes just as hard when I specify a .otf filename.

**Comment by frankrolf** [comment]
The reason for this probably is that cubic input data is expected.
tx is trying to convert it to T1 and then crashes:
```
Downloads $ tx -t1 LSBTTFBug.ufo                                                         [20:06:26]
tx: --- LSBTTFBug.ufo
tx: (ufr) Warning: No version specified in source UFO font.
[1]    1561 segmentation fault  tx -t1 LSBTTFBug.ufo
```

I would say this proves no involvement of the afdko 😅

**Comment by justvanrossum** [comment]
I'm confused, `tx` is not part of afdko?

**Comment by frankrolf** [comment]
The fact that `tx` crashes when presented with quadratic source data means that it cannot be involved in creating a TTF binary. It must come from somewhere else.

**Comment by anthrotype** [comment]
ufo2fdk?

**Comment by justvanrossum** [comment]
It's most likely a RoboFont bug.

**Comment by typemytype** [comment]
ufo2fdk sets it to the left margin of the shape not the most left point... 

https://github.com/robotools/ufo2fdk/blob/99a01d340de3ef3c3d86708c45afebd682daab00/Lib/ufo2fdk/outlineOTF.py#L442

**Comment by ryanbugden** [comment]
Thanks to everyone here and https://github.com/fonttools/fonttools/issues/3072 for your attention to this issue!

**Comment by ryanbugden** [comment]
Fixed in RoboFont Beta 2304031522.
